mypy/errorcodes.py

```diff
@@ -293,5 +293,11 @@ def __hash__(self) -> int:
     "General",
 )
 
+DEPRECATED: Final = ErrorCode(
+    "deprecated",
+    "Warn when importing or using deprecated (overloaded) functions, methods or classes",
+    "General",
```

Comment:
I think this should be `default_enabled=False` (at least for now). Also you must update the docs to include this error code (this is why the docs build failed).

Reply:
When you say "for now", what kind of timeline do you have in mind for turning it on by default? I see that it will be somewhat painful for users who now get these deprecation warnings, but that will also be the case if we turn it on in the future.

It might be nice to present deprecations as warnings by default (i.e., they don't break the type check), but provide an option for users who want any use of deprecated functionality to fail CI.

Reply:
If we turn it on in the future people will be at least somewhat ready.

Mypy doesn't have warnings, it is either an `error` or `note`. That said, `note` may be fine (since notes result in exit code 0), but also IIRC there is no way make a specific note an error.

Reply:
I switched from `error` to `note`.  I can still pass the new error code, which makes a difference when using `type: ignore[deprecated]` or other error codes.  However, the error code does not seem to be appended to the message.

Reply:
@tyralla Yes, error codes on notes are not shown by default, you need to add it to `SHOW_NOTE_CODES` in `errors.py`.

Reply:
Thanks, done. I added three tests which cover the different modes (no reports, notes, errors).